client/state/reader/streams/reducer.js

```diff
@@ -52,7 +52,14 @@ export const pendingItems = ( state = [], action ) => {
 			const { posts } = action.payload;
 			const postKeys = posts.map( keyForPost );
 
-			return uniqBy( postKeys, keyToString );
+			const newState = uniqBy( postKeys, keyToString );
+			if ( newState.length !== state.length ) {
+				return newState;
+			}
+			const hasChanges = some( newState, ( key, idx ) => ! keysAreEqual( key, state[ idx ] ) );
```

Comment:
just found something cool.  [lodash.isEqualWith](https://lodash.com/docs/4.17.5#isEqualWith).

Means we could write:
```js
const hasChanges = ! isEqualWiths( state, newState, keysAreEqual );
```

Reply:
that... didn't work. struggling to understand why though. maybe spurious props on the keys? key equivalence is a bit weird...